test/ExtendedXmlSerializer.Tests/ReportedIssues/Issue141Tests.cs

```diff
@@ -0,0 +1,82 @@
+﻿using ExtendedXmlSerializer;
+using ExtendedXmlSerializer.Configuration;
+using ExtendedXmlSerializer.ExtensionModel;
+using ExtendedXmlSerializer.ExtensionModel.Content;
+using ExtendedXmlSerializer.ExtensionModel.Xml;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace ExtendedXmlSerializer.Tests.ReportedIssues
+{
+    public class Issue141Tests
+    {
+        [Fact]
+        public void ShouldPreserveNullStringValueIfDefaultIsNotNull()
+        {
+            var config = new ConfigurationContainer();
+            config.Emit(EmitBehaviors.Assigned); //no matter what we put here
+            var serializer = config.Create();
+
+            string xml = serializer.Serialize(new ClassWithDefautString() { Name = null, SubNode = null });
+            var deserialized = serializer.Deserialize<ClassWithDefautString>(xml);
+
+            deserialized.Name.Should().BeNull();    //fail
+        }
+
+        [Fact]
+        public void ShouldPreserveNullObjectValueIfDefaultIsNotNull()
+        {
+            var config = new ConfigurationContainer();
+            config.Emit(EmitBehaviors.Assigned); //no matter what we put here
+            var serializer = config.Create();
+
+            string xml = serializer.Serialize(new ClassWithDefautString() { Name = null, SubNode = null });
+            var deserialized = serializer.Deserialize<ClassWithDefautString>(xml);
+
+            deserialized.SubNode.Should().BeNull();    //fail
+        }
+
+        [Fact]
+        public void ShouldPreserveNullAttributeValueIfDefaultIsNotNull()
```

Comment:
OK... looking for some more context around this one.  Why should the `Attribute` be null if it has a default value?  Are you saying the value should not be emitted if it is the same as its default?  If that is the case, there is `EmitBehaviors.NotDefault`.  `Assigned` will emit if it is assigned (not null or `default(T)` in the case of value types).

Reply:
Sorry, my mistake. This is exactly why I should not code late hours :)
I meant line 50 to be:
```csharp 
string xml = serializer.Serialize(new ClassWithDefautString() { Attribute = null, SubNode = null });
```
so similarly to nodes, assigned null value will be  preserved in attributes as well.

Reply:
Ah no worries.  That restores my sanity and makes my life easier. :)  I was sort of wracking on how to approach this.  To add to the chaos, I had my understanding of `Assigned` and `NotDefault` flipped.  So `Assigned` will emit only when the value is not the default value assigned in the constructor.  `NotDefault` will emit when it is not the framework default.

Reply:
Alright, I have done some spelunking here.  `DataContractSerializer` doesn't even support attributes, at least [from what I can see](https://stackoverflow.com/questions/591907/how-can-you-control-net-datacontract-serialization-so-it-uses-xml-attributes-in), so if you have some way of testing how null attributes are emitted with this serializer, then I am all ears.

As for XmlSerializer, this is the outputs from the test above:
```
<?xml version="1.0" encoding="utf-8"?>
<ClassWithDefaultString xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<Name>Unnamed</Name>
	<SubNode>
		<Name>UnnamedSubclass</Name>
	</SubNode>
</ClassWithDefaultString>
```

So, it doesn't emit a null value, which results in the `Attribute` value being set to `Unset`, breaking expected functionality here.

This is a tricky one, as it really rubs up against one of Xml's few limitations.  I am not sure we will be able to get this one to work, unless you have a magic trick that I do not know about. :)